scri/bms_transformations.py

```diff
@@ -0,0 +1,829 @@
+        in terms of the untransformed Ylms. Defaults to 1e-15.
+    """
+    if ell_max is None:
+        ell_max = Lorentz.ell_max
+    n_theta = 2 * ell_max + 1; n_phi = n_theta;
+        
+    transformed_S = transform_function(S, Lorentz, ell_max, tol)
+    
+    transformed_S_S2 = spinsfast.salm2map(transformed_S, 0, ell_max, n_theta, n_phi)
+    K_S2 = compute_conformal_factor(Lorentz, n_theta, n_phi)
+
+    S_prime_S2 = K_S2**(-1)*transformed_S_S2
+
+    return spinsfast.map2salm(S_prime_S2, 0, ell_max)
+    
+class Lorentz_transformation:
+    def __init__(self, **kwargs):
+        self.ell_max = copy.deepcopy(kwargs.pop("ell_max", 30))
+        (
+            rotation,
+            boost,
+            supertranslation,
+        ) = _process_transformation_kwargs(self.ell_max, **kwargs)
+        self.rotation = copy.deepcopy(rotation)
+        self.boost = copy.deepcopy(boost)
+
+        self.order = copy.deepcopy(kwargs.pop("order", ['rotation','boost']))
+        if 'supertranslation' in self.order:
+            del self.order[self.order.index('supertranslation')]
+        for transformation in ['rotation','boost']:
+            if transformation not in self.order:
+                self.order.append(transformation)
+
+    def __repr__(self):
+        Lorentz_output = {}
+        for transformation in self.order:
+            if transformation == 'rotation':
+                Lorentz_output[transformation] = self.rotation
+            elif transformation == 'boost':
+                Lorentz_output[transformation] = self.boost
+
+        return f'Lorentz_transformation(\n\t{self.order[0]}={Lorentz_output[self.order[0]]}\n\t{self.order[1]}={Lorentz_output[self.order[1]]}\n)'
+
+    def __str__(self):
+        Lorentz_output = {}
+        for transformation in self.order:
+            if transformation == 'rotation':
+                Lorentz_output[transformation] = self.rotation
+            elif transformation == 'boost':
+                Lorentz_output[transformation] = self.boost
+
+        return f'Lorentz_transformation(\n\t{self.order[0]}={Lorentz_output[self.order[0]]}\n\t{self.order[1]}={Lorentz_output[self.order[1]]}\n)'
+    
+    def copy(self):
+        return Lorentz_transformation(rotation=self.rotation,
+                                      boost=self.boost,
+                                      order=self.order,
+                                      ell_max=self.ell_max)
+
+    def reorder(self, output_order):
+        """Reorder a Lorentz transformation.
+        """
+        if not ('rotation' in output_order and\
+                'boost' in output_order):
+            raise ValueError('Not enough transformations')
+        
+        if self.order == output_order:
+            return self
+        else:
+            L_spin_matrix = Lorentz_to_spin_matrix(self)
+            L_reordered = spin_matrix_to_Lorentz(L_spin_matrix, output_order=output_order)
+            return L_reordered
```

Comment:
If the order matches, we get back the original object, whereas if the order differs, we get back a new object. If we get back the same object, mutating the result will mutate the original; but not if we get back a different object. Maybe the first case should make a copy of itself instead of return self?